src/Avalonia.Controls/Shapes/Arc.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using Avalonia.Media;
+
+namespace Avalonia.Controls.Shapes
+{
+    public class Arc : Shape
+    {
+        /// <summary>
+        /// Defines the <see cref="StartAngle"/> property.
+        /// </summary>
+        public static readonly StyledProperty<double> StartAngleProperty =
+            AvaloniaProperty.Register<Arc, double>(nameof(StartAngle), 0.0);
+
+        /// <summary>
+        /// Defines the <see cref="SweepAngle"/> property.
+        /// </summary>
+        public static readonly StyledProperty<double> SweepAngleProperty =
+            AvaloniaProperty.Register<Arc, double>(nameof(SweepAngle), 0.0);
+
+        static Arc()
+        {
+            StrokeThicknessProperty.OverrideDefaultValue<Arc>(1);
+            AffectsGeometry<Arc>(BoundsProperty, StrokeThicknessProperty, StartAngleProperty, SweepAngleProperty);
+        }
+
+        /// <summary>
+        /// Gets or sets the angle at which the arc starts, in degrees.
+        /// </summary>
+        public double StartAngle
+        {
+            get { return GetValue(StartAngleProperty); }
+            set { SetValue(StartAngleProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the angle at which the arc ends relative to the start angle, in degrees.
+        /// </summary>
+        public double SweepAngle
+        {
+            get { return GetValue(SweepAngleProperty); }
+            set { SetValue(SweepAngleProperty, value); }
+        }
+        
+        protected override Geometry CreateDefiningGeometry()
+        {
+            double angle1 = DegreesToRad(StartAngle);
+            double angle2 = angle1 + DegreesToRad(SweepAngle);
+            
+            double startAngle = Math.Min(angle1, angle2);
+            double sweepAngle = Math.Max(angle1, angle2);
+
+            double normStart = RadToNormRad(startAngle);
+            double normEnd = RadToNormRad(sweepAngle);
+
+            var rect = new Rect(Bounds.Size);
+
+            if ((normStart == normEnd) && (startAngle != sweepAngle)) //complete ring
+            {
+                return new EllipseGeometry(rect.Deflate(StrokeThickness / 2));
+            }
+            else if (SweepAngle == 0)
+            {
+                return new StreamGeometry();
+            }
+            else //partial arc
+            {
+                var deflatedRect = rect.Deflate(StrokeThickness / 2);
+
+                double centerX = rect.Center.X;
```

Comment:
Use `var`

Reply:
var was originally added to make Linq usage more sane. For basic types line this it can actually make code less readable.

Reply:
I think this is more about code style. If the code base usually uses `var` that should be followed.

Reply:
It is entirely possible to overuse var. In this case it isn't obvious if the type is float or double unless you have a lot of experience with the objects. This is especially true for those of us that use WinUI where Microsoft is changing to float. There is no reason to hide the double type in this case, it isnt too verbose.

Reply:
I dont know how you can disagree with this. Almost any discussion, article or commentary on the use of var will back up my points.

Reply:
_I_ certainly can disagree with it, and when I've seen it discussed on twitter etc, it appears to me to entirely come down to taste. Personally I much prefer `var`.

Reply:
If this direction is set by you, I can't change it. You founded the code and can firmly set the standard for things like this. However, var is not intended for use everywhere like in a non-typesafe scripting language. It would be wise to let developers use it as they see best and error on the side of not-using var to ensure types are readable and clear. I've given examples for this case (float vs double). The arguments of my viewpoint are objectively stronger than simply requiring var for consistency and code style.

Reply:
I agree, I wouldn't have flagged usage of non-`var` in this PR personally.

Reply:
...so is there a final decision on this yet?

Reply:
@Splitwirez use var as i suggested. That's how our codebase is.

Reply:
I strongly recommend you change this requirement as it reduces readability in cases like this and hides information. I can even prove it violates the design intent of C# and Microsofts guidelines.

 If I commit code to the repo expect another conversation about it. Otherwise, 

 * https://www.researchgate.net/profile/Pierre-Akiki/publication/328999651_To_var_or_not_to_var_how_do_C_developers_use_and_misuse_implicit_and_explicit_typing/links/5c3b160c299bf12be3c4334f/To-var-or-not-to-var-how-do-C-developers-use-and-misuse-implicit-and-explicit-typing.pdf
 * https://intellitect.com/when-to-use-and-not-use-var-in-c/

These arguments are immensely stronger than those for consistency.

Reply:
Let's close this conversation about style guides in this PR. 
In general I prefer consistency with whatever approach was actually used and whatever advantages or disadvantages it has. Still code style question does not affects neither Avalonia behavior nor user applications behavior.
So, if we want to discuss code style questions and enforce some specific rules (even with https://github.com/dotnet/format) we should continue in separated discussion.

And as for this PR, we should ignore this problem and unblock contributors.